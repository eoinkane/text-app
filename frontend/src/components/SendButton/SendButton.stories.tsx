import React from "react";
import { action } from "@storybook/addon-actions";
import SendButtonComponent from "./index";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";

export default {
  title: "SendButton",
  component: SendButtonComponent,
};

export const SendButton = () => {
  return (
    <Container style={{ padding: "4vh 4vw" }}>
      <Typography style={{ textAlign: "center", paddingBottom: 10 }}>
        Send Button
      </Typography>
      <Container
        style={{
          backgroundColor: "#f5f5f5",
          margin: "20px 0",
        }}
      >
        <SendButtonComponent onClick={action("button-click")} />
      </Container>
    </Container>
  );
};
