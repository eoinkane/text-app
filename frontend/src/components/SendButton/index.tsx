import React, { SyntheticEvent } from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import { IconButton } from "@material-ui/core";
import SendIcon from "@material-ui/icons/Send";

interface SendButtonProps {
  onClick: (event: SyntheticEvent) => void;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      padding: "10px 0",
    },
    button: {
      backgroundColor: "rgb(124, 224, 62)!important",
    },
  })
);

const SendButton: React.FC<SendButtonProps> = ({ onClick }) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <IconButton onClick={onClick} className={classes.button}>
        <SendIcon />
      </IconButton>
    </div>
  );
};

export default SendButton;
