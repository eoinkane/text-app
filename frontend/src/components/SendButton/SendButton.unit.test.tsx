import React from "react";
import { shallow } from "enzyme";
import SendButton from "./index";
import IconButton from "@material-ui/core/IconButton";
import SendIcon from "@material-ui/icons/Send";

describe("SendButton", () => {
  it("Renders ", () => {
    const sendButton = shallow(<SendButton onClick={() => {}} />);
    expect(sendButton).toMatchSnapshot();
  });

  it("Renders Icon Button", () => {
    const sendButton = shallow(<SendButton onClick={() => {}} />);
    expect(sendButton).toMatchSnapshot();

    expect(sendButton.find(IconButton).length).toBe(1);
  });

  it("Renders Send Icon", () => {
    const sendButton = shallow(<SendButton onClick={() => {}} />);
    expect(sendButton).toMatchSnapshot();

    expect(sendButton.find(IconButton).dive().find(SendIcon).length).toBe(1);
  });

  it("Test click event", () => {
    const mockCallBack = jest.fn();

    const sendButton = shallow(<SendButton onClick={mockCallBack} />);
    sendButton.find(IconButton).simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
