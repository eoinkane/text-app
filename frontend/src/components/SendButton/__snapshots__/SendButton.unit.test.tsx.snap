// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendButton Renders  1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(IconButton))
    className="makeStyles-button-2"
    onClick={[Function]}
  >
    <SendIcon />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`SendButton Renders Icon Button 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(IconButton))
    className="makeStyles-button-2"
    onClick={[Function]}
  >
    <SendIcon />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`SendButton Renders Send Icon 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(IconButton))
    className="makeStyles-button-2"
    onClick={[Function]}
  >
    <SendIcon />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;
